users/admin.py

```diff
@@ -20,9 +20,31 @@
 Application = get_application_model()
 
 
+class UserSocialAuthInline(admin.StackedInline):
+    model = UserSocialAuth
+    readonly_fields = ['provider', 'uid', 'get_extra_data']
+    exclude = ['extra_data']
+    extra = 0
+
+    def get_extra_data(self, obj=None):
+        if not obj or not obj.extra_data:
+            return ''
+
+        return mark_safe(f'<pre>{json.dumps(obj.extra_data, indent=4)}</pre>')
```

Comment:
The result of `json.dumps` should probably be HTML escaped. Otherwise it could somehow be possible to inject some HTML, including javascript, into the page.

Reply:
Very true.

Reply:
Now it got fixed in the second commit. It could be correctly already in the first commit where that code is introduced.